src/app/submissions/submissions.component.ts

```diff
@@ -126,7 +129,11 @@ export class SubmissionsComponent implements OnInit, AfterViewChecked, OnDestroy
   }
 
   setupTable() {
-    this.submissions.filterPredicate = composeFilterFunctions([ filterDropdowns(this.filter), filterSpecificFields([ 'parent.name' ]) ]);
+    this.submissions.filterPredicate = composeFilterFunctions(
+      [
+        filterSpecificFieldsByWord([ 'parent.name' ]),
+        filterDropdowns(this.filter), filterSpecificFields([ 'parent.name' ])
```

Comment:
You need to remove `filterSpecificFields`.  It's conflicting with `filterSpecificFieldsByWord`.